lib/ace/ext/language_tools.js

```diff
@@ -51,6 +55,12 @@ var keyWordCompleter = {
 
 var snippetCompleter = {
     getCompletions: function(editor, session, pos, prefix, callback) {
+        // exit if we're inside an html tag
+        var token = session.getTokenAt(pos.row, pos.column);
```

Comment:
This doesn't seem quite right, there can be snippets that are meant to be used inside tags, and the snippets that are not meant to be used, tab key should be disabled too. Perhaps we need to modify getActiveScopes instead?

Reply:
Yes, I think you're right. I'm thinking we need a new scope for when we're in an HTML tag, which is what I ended up doing with my dev version. This may be tricky as the snippet files only contain one scope though.

Reply:
every snippet can contain it's out scope too, the one from file is the default.

Reply:
Great, so instead of returning an empty array we could change the scope to `html-tag`.  Sound good?

Reply:
just add `scope html-tag` after `snippet name`